### PR TITLE
Support names and usernames in author metadata schema

### PR DESCRIPTION
**Pull request by Mandrenkov**
**Title:**

Adds support for specifying authors in demo metadata via full name or PennyLane profile username.

**Summary:**

The following metadata file is now accepted by the JSON validator:
```js
{
    "authors": [
        {
            "id": "jack_ceroni"
        },
        {
            "username": "mikhail"
        },
        {
            "name": "Mikhail Andrenkov"
        }
    ]
    // other fields ...
}
```
Conversely, the metadata file below is _not_ accepted:
```js
{
    "authors": [
        {},  // no matches
        {
            "username": "mikhail",
            "name": "Mikhail Andrenkov"  // extra field
        },
        {
            "username": "mi"  // too short
        },
        {
            "username": "abcdefghijklmnopqrstu"  // too long
        },
        {
            "name": "M"  // too short
        }
    ]
    // other fields ...
}
```

**Relevant references:**

N/A

**Possible Drawbacks:**

1. If this PR is merged but the decision is later reverted, there will be some extraneous metadata versioning.

**Related GitHub Issues:**

N/A

### CODE REVIEW

**Comment by ikurecic** [comment]
Thanks a lot for the tag, @Mandrenkov .
This might be unrelated, but I'd like to understand everything to make sure I can follow the setup correctly — in the first example (copied below), what is the `id` field? 

> ```js
> {
>     "authors": [
>         {
>             "id": "jack_ceroni"
>         },
>         {
>             "username": "mikhail"
>         },
>         {
>             "name": "Mikhail Andrenkov"
>         }
>     ]
>     // other fields ...
> }
> ```

**Comment by Mandrenkov** [comment]
Not a problem, @ikurecic!

> in the first example (copied below), what is the `id` field?

The `id` field is part of the [author metadata format](https://github.com/PennyLaneAI/qml/blob/master/metadata_schemas/objects/author.schema.0.1.0.json) that is currently used by all demos in this repository. My understanding is that it refers to the file stem of an author in [this directory](https://github.com/PennyLaneAI/qml/tree/master/_static/authors) which contains `.txt` files that have the author's name and bio.

As a whole, the example is meant to show that this PR adds support for the new author metadata format _while retaining support for the old format_. Eventually, we'll want to drop support for the old format too, but that is later down the line.

**Comment by ikurecic** [comment]
I see what you mean, @Mandrenkov — it didn't cross my mind that this would still refer to the old/current system. That's great to know and it makes sense to me, thank you!

**Review by ikurecic — APPROVED** [review]
The result makes a lot of sense to me, but I can't check the details of the implementation without spending a bit of time understanding the existing structure. So I'm approving the described result here. Thank you!

File: .github/workflows/validate-demo-metadata.yml [hunk 1]
```diff
@@ -101,7 +101,7 @@ jobs:
           METADATA_FILE_LIST: ${{ needs.generate-metadata-file-list.outputs.metadata_files }}
         run: |
           cd metadata_schemas
-          ${{ steps.poetry.outputs.bin }} run check-jsonschema -v --traceback-mode full --schemafile demo.metadata.schema.0.1.0.json $METADATA_FILE_LIST
+          ${{ steps.poetry.outputs.bin }} run check-jsonschema -v --traceback-mode full --schemafile demo.metadata.schema.0.1.1.json $METADATA_FILE_LIST
```
Comment (doctorperceptron) [inline .github/workflows/validate-demo-metadata.yml:104]: :partying_face:

**Review by doctorperceptron — APPROVED** [review]
LGTM!

**Pull request merged**